libpal/libpal.h

```diff
@@ -25,6 +25,16 @@
 #ifndef LIBPAL_H
 #define LIBPAL_H
 
+/* 
+ * Make sure that extern functions build for
+ * C and C++ 
+*/
+#ifdef __cplusplus
+#define __extern extern "C"
```

Comment:
Instead of defining these this way at each function declaration, can we just do a single block?

```
#ifdef __cplusplus
extern "C" {
#endif

<function declarations>

#ifdef __cplusplus
}
#endif
```

Reply:
Yeah that makes sense.